src/qt/dcclient.cpp

```diff
@@ -67,15 +89,19 @@ wxClientDCImpl::wxClientDCImpl( wxDC *owner )
     : wxWindowDCImpl( owner )
 {
     m_window = NULL;
+    m_qtPainter = NULL;
 }
 
 wxClientDCImpl::wxClientDCImpl( wxDC *owner, wxWindow *win )
     : wxWindowDCImpl( owner )
 {
     m_window = win;
 
-    QPicture *pic = win->QtGetPicture();
-    m_ok = m_qtPainter->begin( pic );
+    m_qtPainter = new QPainter();
```

Comment:
It looks like this is going to be leaked. The code is rather confusing but AFAICS `m_qtPainter` is reset to NULL in the base class dtor, so the comment at the end of this class dtor is wrong. Isn't it?

Reply:
Thankfully the `m_qtPainter` member will not be leaked. The inheritance structure of the classes here is as follows:

`wxClientDCImpl` and `wxPaintDCImpl` both inherit from `wxWindowDCImpl`
`wxWindowDCImpl` inherits from `wxQtDCImpl`

The `m_qtPainter` member is declared in the base class (`wxQtDCImpl`), as well as being deleted in its destructor. Thanks to the deletion taking place in the base, we can be certain that the `QPainter` will be deleted whenever an instance of one of these classes goes out of scope.